lab2/app.js

```diff
@@ -0,0 +1,205 @@
+// single state store
+class Store {
+    constructor(storage) {
+        this.storage = storage; // assuming local storage will be passed in as storage
+        // these are the key name you can use in Store
+        this.CART_KEY = 'CART';
+        this.QUEUE_KEY = 'QUEUE';
+        this.FOODS_KEY = 'FOODS';
+    }
+
+    // you can get item by store.cartItems
+    get cartItems() {
+        return JSON.parse(this.storage.getItem(this.CART_KEY));
+    }
+
+    // to call setter, simply "assign" like store.cartItems = something
+    set cartItems(cartItems) {
+        this.storage.setItem(this.CART_KEY, JSON.stringify(cartItems));
+    }
+
+    get queue() {
+        return JSON.parse(this.storage.getItem(this.QUEUE_KEY));
+    }
+
+    set queue(queue) {
+        this.storage.setItem(this.QUEUE_KEY, JSON.stringify(queue));
+    }
+
+    get foods() {
+        return JSON.parse(this.storage.getItem(this.FOODS_KEY));
+    }
+
+    set foods(foods) {
+        this.storage.setItem(this.FOODS_KEY, JSON.stringify(foods));
+    }
+}
+
+class Cart {
+    // take dom element into JavaScript class to attachment events
+    constructor(root, store) {
+        this.root = root;
+        this.store = store;
+        this.items = this.store.cartItems;
+        this.init();
+    }
+
+    init() {
+        // Render a list of items under root element
+        this.render();
+        // TODO: attach remove cart items to rendered HTML
+    }
+
+    destroy() {
+        // TODO: remove all the events attached from init
+    }
+
+    // remove an item from shopping cart
+    removeItem(item) {
+        // TODO: logic to remove an item from cart
+        // call render method when the item is removed to update view
+        this.render();
+    }
+
+    placeOrder() {
+        // add item to statuses in store as status "in progress"
+    }
+
+    // render a list of item under root element
+    render() {
+        console.log(this.store.cartItems);
+        let tbody = this.root.querySelector('tbody');
+        // using innerHTML to render a list of table row item under tbody
+        tbody.innerHTML = this.renderListAsHTML(this.store.cartItems)
+        let deleteButtons = this.root.querySelectorAll('.delete-button');
+       
+       
+        let fakeList = this.store.cartItems;
+       
+        for (var i = 0; i < deleteButtons.length; i++) {
+            let deleteBtn = deleteButtons[i];
+            let scopedFakeList = fakeList;
+            deleteBtn.addEventListener('click', () => {
+                debugger;
+                
+                this.store.cartItems.removeItem();
```

Comment:
there is no remove item method under this.store.cartItems.removeItem();

You will need to do your own array mutation